sentry/src/main/java/io/sentry/instrumentation/file/SentryFileOutputStream.java

```diff
@@ -0,0 +1,196 @@
+    this(init(name != null ? new File(name) : null, false, null));
+  }
+
+  public SentryFileOutputStream(String name, boolean append) throws FileNotFoundException {
+    this(init(name != null ? new File(name) : null, append, null));
+  }
+
+  public SentryFileOutputStream(File file) throws FileNotFoundException {
+    this(init(file, false, null));
+  }
+
+  public SentryFileOutputStream(File file, boolean append) throws FileNotFoundException {
+    this(init(file, append, null));
+  }
+
+  public SentryFileOutputStream(FileDescriptor fdObj) {
+    this(init(fdObj, null), fdObj);
+  }
+
+  private SentryFileOutputStream(
+    @NotNull FileOutputStreamInitData data,
+    @NotNull FileDescriptor fd
+  ) {
+    super(fd);
+    file = null;
+    currentSpan = data.span;
+    delegate = data.delegate;
+  }
+
+  private SentryFileOutputStream(
+    @NotNull FileOutputStreamInitData data
+  ) throws FileNotFoundException {
+    super(data.file, data.append);
+    currentSpan = data.span;
+    file = data.file;
+    delegate = data.delegate;
+  }
+
+  private static FileOutputStreamInitData init(
+    @Nullable File file,
+    boolean append,
+    @Nullable FileOutputStream delegate
+  ) throws FileNotFoundException {
+    final ISpan span = startSpan();
+    if (delegate == null) {
+      delegate = new FileOutputStream(file);
+    }
+    return new FileOutputStreamInitData(file, append, span, delegate);
+  }
+
+  private static FileOutputStreamInitData init(
+    @NotNull FileDescriptor fd,
+    @Nullable FileOutputStream delegate
+  ) {
+    final ISpan span = startSpan();
+    if (delegate == null) {
+      delegate = new FileOutputStream(fd);
+    }
+    // TODO: it's only possible to get filename from FileDescriptor via reflection AND when it's
+    // TODO: running on Android, should we do that?
+    return new FileOutputStreamInitData(null, false, span, delegate);
+  }
+
+  private static @Nullable ISpan startSpan() {
+    final ISpan parent = Sentry.getSpan();
+    return parent != null ? parent.startChild("file.write") : null;
+  }
+
+  @Override public void write(int b) throws IOException {
+    try {
+      delegate.write(b);
+      byteCount++;
+    } catch (IOException exception) {
+      spanStatus = SpanStatus.INTERNAL_ERROR;
+      throw exception;
+    }
+  }
+
+  @Override public void write(byte[] b) throws IOException {
+    try {
+      delegate.write(b);
+      byteCount += b.length;
+    } catch (IOException exception) {
+      spanStatus = SpanStatus.INTERNAL_ERROR;
+      throw exception;
+    }
+  }
+
+  @Override public void write(byte[] b, int off, int len) throws IOException {
+    try {
+      delegate.write(b, off, len);
+      byteCount += len;
+    } catch (IOException exception) {
+      spanStatus = SpanStatus.INTERNAL_ERROR;
+      throw exception;
+    }
+  }
+
+  @Override public void close() throws IOException {
+    try {
+      delegate.close();
```

Comment:
this class has plenty of very similar methods compared to `SentryFileInputStream`, can we do some DRY?

Reply:
yeah, thought of this already, will try to make smthng up

Reply:
Extracted the logic into a `FileIOSpanManager` - not a big fan of having a `manager` in the name, but couldn't come up with anything better